--- a/lib/platform/util/atomic.h	2015-04-07 02:49:07.000000000 +1000
+++ b/lib/platform/util/atomic.h	2015-07-13 20:08:34.746638285 +1000
@@ -29,7 +29,7 @@
 ///////////////////////////////////////////////////////////////////////////
 static inline long atomic_inc(volatile long* pAddr)
 {
-#if defined(HAS_BUILTIN_SYNC_ADD_AND_FETCH)
+#if defined(HAS_BUILTIN_SYNC_ADD_AND_FETCH) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 7))
   return __sync_add_and_fetch(pAddr, 1);
 
 #elif defined(__ppc__) || defined(__powerpc__) // PowerPC
@@ -49,14 +49,14 @@
 #elif defined(__arm__) && !defined(__ARM_ARCH_5__)
   register long val;
   asm volatile (
-    "dmb      ish            \n" // Memory barrier. Make sure all memory accesses appearing before this complete before any that appear after
+    "dmb\n" // Memory barrier. Make sure all memory accesses appearing before this complete before any that appear after
     "1:                     \n"
     "ldrex   %0, [%1]       \n" // (val = *pAddr)
     "add     %0,  #1        \n" // (val += 1)
     "strex   r1,  %0, [%1]      \n"
     "cmp     r1,   #0       \n"
     "bne     1b             \n"
-    "dmb     ish            \n" // Memory barrier.
+    "dmb\n" // Memory barrier.
     : "=&r" (val)
     : "r"(pAddr)
     : "r1"
