diff --git a/src/settings/plugins/keyfile/reader.c b/src/settings/plugins/keyfile/reader.c
index 6ddf50d..11b70dc 100644
--- a/src/settings/plugins/keyfile/reader.c
+++ b/src/settings/plugins/keyfile/reader.c
@@ -1109,7 +1109,6 @@
 {
 	GKeyFile *key_file;
 	struct stat statbuf;
-	gboolean bad_owner, bad_permissions;
 	NMConnection *connection = NULL;
 	NMSettingConnection *s_con;
 	NMSetting *setting;
@@ -1123,16 +1122,6 @@
 	if (stat (filename, &statbuf) != 0 || !S_ISREG (statbuf.st_mode)) {
 		g_set_error_literal (error, KEYFILE_PLUGIN_ERROR, 0,
 		                     "File did not exist or was not a regular file");
-		return NULL;
-	}
-
-	bad_owner = getuid () != statbuf.st_uid;
-	bad_permissions = statbuf.st_mode & 0077;
-
-	if (bad_owner || bad_permissions) {
-		g_set_error (error, KEYFILE_PLUGIN_ERROR, 0,
-		             "File permissions (%o) or owner (%d) were insecure",
-		             statbuf.st_mode, statbuf.st_uid);
 		return NULL;
 	}
 